userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -377,4 +466,131 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
     return SCAP_TIMEOUT;
 }
 
+engine::runsc_result engine::runsc(char *argv[])
+{
+	runsc_result res = {0};
+	int pipefds[2];
+
+	int ret = pipe(pipefds);
+	if(ret)
+	{
+		return res;
+	}
+
+	pid_t pid = vfork();
+	if(pid > 0)
+	{
+		char line[max_line_size];
+		int status;
+		
+		::close(pipefds[1]);
+		wait(&status);
+		if(status)
```

Comment:
Shouldn't we be using something like `if (!WIFEXITED(status) || WEXITSTATUS(status) != 0)`.
I mean, we should check:
* if the command was killed by a signal
* if the command exited normally but with a non-zero return status

Reply:
Agreed and added